jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -1055,10 +1055,9 @@ void java_object_factoryt::gen_nondet_struct_init(
     // passes can easily recognise leaves no uninitialised state behind.
 
     // This code mirrors the `remove_java_new` pass:
-    null_message_handlert nullout;
     exprt zero_object=
       zero_initializer(
-        struct_type, source_locationt(), ns, nullout);
+        struct_type, source_locationt(), ns, null_message_handler);
```

Comment:
This code should likely use the exception-throwing variant of `zero_initializer` as the value isn't even sanity-checked afterwards. (And that variant would not require a message handler.)